drivers/sensor/bmi270/bmi270.c

```diff
@@ -0,0 +1,676 @@
+
+		ret = reg_read(BMI270_REG_PWR_CTRL, &pwr_ctrl, 1, dev);
+		if (ret != 0) {
+			return ret;
+		}
+	}
+
+	if (odr) {
+		odr_bits = gyr_odr_to_reg(odr);
+		gyr_conf = BMI270_SET_BITS_POS_0(gyr_conf, BMI270_GYR_ODR,
+						 odr_bits);
+
+		/* If odr_bits is 0, implies that the sampling frequency is
+		 * 0Hz or invalid too.
+		 */
+		if (odr_bits) {
+			pwr_ctrl |= BMI270_PWR_CTRL_GYR_EN;
+		} else {
+			pwr_ctrl &= ~BMI270_PWR_CTRL_GYR_EN;
+		}
+
+		/* If the Sampling frequency (odr) >= 100Hz, enter performance
+		 * mode else, power optimized. This also has a consequence for
+		 * the OSR
+		 */
+		if (odr_bits >= BMI270_GYR_ODR_100_HZ) {
+			gyr_conf = BMI270_SET_BITS(gyr_conf,
+					BMI270_GYR_FILT,
+					BMI270_GYR_FILT_PERF_OPT);
+			gyr_conf = BMI270_SET_BITS(gyr_conf,
+					BMI270_GYR_FILT_NOISE,
+					BMI270_GYR_FILT_NOISE_PERF_OPT);
+		} else {
+			gyr_conf = BMI270_SET_BITS(gyr_conf,
+					BMI270_GYR_FILT,
+					BMI270_GYR_FILT_PWR_OPT);
+			gyr_conf = BMI270_SET_BITS(gyr_conf,
+					BMI270_GYR_FILT_NOISE,
+					BMI270_GYR_FILT_NOISE_PWR_OPT);
+		}
+
+		dev->gyr_odr = odr_bits;
+	}
+
+	if (osr) {
+		/* osr->val2 should be unused */
+		switch (osr->val1) {
+		case 4:
+			osr_bits = BMI270_GYR_BWP_OSR4;
+			break;
+		case 2:
+			osr_bits = BMI270_GYR_BWP_OSR2;
+			break;
+		default:
+			osr_bits = BMI270_GYR_BWP_NORM;
+			break;
+		}
+
+		gyr_conf = BMI270_SET_BITS(gyr_conf, BMI270_GYR_BWP,
+					   osr_bits);
+	}
+
+	if (odr || osr) {
+		ret = reg_write(BMI270_REG_GYR_CONF, &gyr_conf, 1, dev);
+		if (ret != 0) {
+			return ret;
+		}
+
+		/* Assuming we have advance power save enabled */
+		k_usleep(BMI270_TRANSC_DELAY_SUSPEND);
+
+		pwr_ctrl &= BMI270_PWR_CTRL_MSK;
+		ret = reg_write(BMI270_REG_PWR_CTRL, &pwr_ctrl, 1, dev);
+		if (ret != 0) {
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+int set_gyro_range(const struct sensor_value *range,
+		   struct bmi270_dev *dev)
+{
+	int ret = 0;
+	uint8_t gyr_range, reg;
+
+	ret = reg_read(BMI270_REG_GYR_RANGE, &gyr_range, 1, dev);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* range->val2 is unused */
+	switch (range->val1) {
+	case 125:
+		reg = BMI270_GYR_RANGE_125DPS;
+		dev->gyr_range = 125;
+		break;
+	case 250:
+		reg = BMI270_GYR_RANGE_250DPS;
+		dev->gyr_range = 250;
+		break;
+	case 500:
+		reg = BMI270_GYR_RANGE_500DPS;
+		dev->gyr_range = 500;
+		break;
+	case 1000:
+		reg = BMI270_GYR_RANGE_1000DPS;
+		dev->gyr_range = 1000;
+		break;
+	case 2000:
+		reg = BMI270_GYR_RANGE_2000DPS;
+		dev->gyr_range = 2000;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	gyr_range = BMI270_SET_BITS_POS_0(gyr_range, BMI270_GYR_RANGE, reg);
+
+	ret = reg_write(BMI270_REG_GYR_RANGE, &gyr_range, 1, dev);
+
+	return ret;
+}
+
+int bmi270_init(const struct device *dev)
+{
+	int ret;
+	struct bmi270_dev *drv_dev = dev->data;
+	uint8_t chip_id;
+	uint8_t soft_reset_cmd;
+	uint8_t init_ctrl;
+	uint8_t msg;
+	uint8_t tries;
+	uint8_t adv_pwr_save;
+
+	drv_dev->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
```

Comment:
Get the bus label and address from a constant device config struct. This will make it possible to enable multiple instances of the driver